polus-csv-statistics-plugin/src/main.py

```diff
@@ -0,0 +1,217 @@
+    parser.add_argument('--statistics', dest='statistics', type=str,
+                        help='Types of statistics to calculate', required=True)
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                        help='Input csv collection to be processed by this plugin', required=True)
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+    
+    # Parse the arguments
+    args = parser.parse_args()
+    statistics = args.statistics.split(',')
+    logger.info('statistics = {}'.format(statistics))
+    inpDir = args.inpDir
+    logger.info('inpDir = {}'.format(inpDir))
+    outDir = args.outDir
+    logger.info('outDir = {}'.format(outDir))
+    
+    # Get a list of all input files
+    csv_files = [f for f in Path(inpDir).iterdir() if f.name.endswith('csv')]
+    
+    # Open each csv files
+    for feat_file in csv_files:
+        fpath = str(feat_file.absolute())
+        out = str(Path(outDir).joinpath(feat_file.name).absolute())
+        with open(fpath,'r') as fr:
+            with open(str(out),'w') as fw:
+                # Read the first line, which should contain headers
+                first_line = fr.readline()
+                headers = first_line.rstrip('\n').split(',')
+                var_ind = {key:val for key,val in enumerate(headers)} # map headers to line positions
+                
+                # If no column is labeled file, throw an error
+                if 'file' not in headers:
+                    ValueError('At least one column must have a header title file.')
+                
+                # Generate the output dictionary template and format string
+                line_dict = {'file': 'NaN'}
+                for key in headers:
+                    if key=='file':
+                        continue
+                    for stat in statistics:
+                        line_dict[key + '_' + stat] = 'NaN'
+                        # Generate the line template
+                line_template = ','.join([('{' + h + '}') for h in line_dict.keys()]) + '\n'
+
+                # Write headers to the new file
+                fw.write(','.join(line_dict.keys()) + '\n')
+
+                # Get the first line of data
+                line = fr.readline()
+
+                # Read each line in the stitching vector
+                fnum = 0
+                fcheck = 0
+                while line:
+                    # Parse the current line as a dictionary
+                    p_line = {var_ind[ind]:val for ind,val in enumerate(line.rstrip('\n').split(','))}
+                    for key,val in p_line.items():
+                        v = get_number(val)
+                        p_line[key] = [v]
+
+                    # Loop through rows until the filename changes
+                    line = fr.readline()
+                    np_line = {var_ind[ind]:val for ind,val in enumerate(line.rstrip('\n').split(','))}
```

Comment:
Can you provide a little more description here? It seems like what you are saying you are while looping over rows but then when I look at this I think you are while looping over columns no? Maybe I've misunderstood what this code is doing. Either way clarifying would be helpful.